packages/components/addon/components/hds/app-footer/status-link.hbs

```diff
@@ -0,0 +1,12 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: MPL-2.0
+}}
+
+<Hds::AppFooter::Link class={{this.classNames}} @href={{this.href}} ...attributes>
+  {{#if this.statusIcon}}
+    <FlightIcon @size="16" @name={{this.statusIcon}} @color={{this.statusIconColor}} />
```

Comment:
I would use the `<Hds::AppFooter::Link` support for an `@icon` argument.
Which leads to the point: should we leave the `@iconPosition` to support this?

In that case, you have for free: 
- icon size and alignment
- you need to "just" override the colors of the icon

Otherwise you have to take care of these problems:
- extra underline between icon and text
- vertical misalignment of the icon

<img width="261" alt="screenshot_3095" src="https://github.com/hashicorp/design-system/assets/686239/25948089-bd20-45e3-9a80-f824d630167a">

Reply:
Re. hiding the `@iconPosition` for the consumers, a simple workaround it would be to use in this component `<Hds::AppFooter::Item>` with a nested `<Hds::Interactive class="hds-app-footer__link">` (same structure as the `<Hds::AppFooter::Link>`)

Reply:
I'm trying to refactor this but have to figure out how to set a custom iconColor value since the Link Inline that the AppFooter::Link uses doesn't support setting an icon color. I'll keep working on it tomorrow. (I have to add an inline style with a custom property or something...)

Reply:
This has been done as part of implementing themes.